src/select/Select.ts

```diff
@@ -0,0 +1,506 @@
+ * @cssprop --omni-select-item-width - Select item width.
+ *
+ * @cssprop --omni-select-item-hover-background-color - Select item hover background color.
+ * @cssprop --omni-select-item-selected-color - Selected item color.
+ * @cssprop --omni-select-item-none-hover - Select item hover.
+ *
+ * @cssprop --omni-select-loading-indicator-width - Select loading indicator width.
+ * @cssprop --omni-select-loading-indicator-height - Select loading indicator height.
+ *
+ */
+@customElement('omni-select')
+export class Select extends OmniFormElement {
+    @query('#select')
+    private _selectElement: HTMLInputElement;
+
+    /**
+     * Selectable items of the select component.
+     * @attr
+     */
+    @property({ type: Array, reflect: true }) items: SelectItems | (() => SelectItems);
+
+    /**
+     * The field of the item to display as one of the selectable options.
+     * @attr
+     */
+    @property({ type: String, reflect: true, attribute: 'display-field' }) displayField: string;
+
+    /**
+     * The id field of the items provided.
+     * @attr
+     */
+    @property({ type: String, reflect: true, attribute: 'id-field' }) idField: string = 'id';
+
+    /**
+     * The render function for each item.
+     */
+    @property({ type: Object, reflect: false }) renderItem: RenderFunction;
+
+    // Internal state properties
+    @state() private _popUp: boolean = false;
+    @state() private _bottomOfScreen: boolean = false;
+    @state() private _mobileWidth: boolean = false;
+
+    constructor() {
+        super();
+        this._sizeCheck();
+    }
+
+    override connectedCallback() {
+        super.connectedCallback();
+        this.addEventListener('click', this._inputClick.bind(this));
+        window.addEventListener('click', this._windowClick.bind(this));
+        window.addEventListener('resize', this._sizeCheck.bind(this));
+    }
+
+    _inputClick() {
+        this._togglePopup();
+    }
+
+    // https://stackoverflow.com/a/39245638
+    // Close the item container when clicking outside the select component.
+    _windowClick(e: Event) {
+        if (e.composedPath() && !e.composedPath().includes(this) && this._popUp) {
+            this._togglePopup();
+        }
+    }
+
+    _controlClick() {
+        this._togglePopup();
+    }
+
+    _togglePopup() {
+        if (this._popUp) {
+            this._popUp = false;
+        } else {
+            this._popUp = true;
+        }
+    }
+
+    // Set the value when a item is clicked
+    async _onItemClick(item: string) {
+        this.value = item;
+
+        await this.updateComplete;
+
+        //https://stackoverflow.com/a/36084475
+        this._selectElement.dispatchEvent(
+            new Event('change', {
+                bubbles: true,
+                composed: true
+            })
+        );
+    }
+
+    // Check the size of the screen to determine how to render items and which control icon to apply
+    async _sizeCheck() {
+        this._heightCheck();
```

Comment:
If this function is async should these not be awaited?

Reply:
Original implementation was async this current implementation does not require it.